src/library/scala/MatchError.scala

```diff
@@ -19,11 +21,11 @@ package scala
  *  @version 1.1, 05/03/2004
  *  @since   2.0
  */
-final class MatchError(obj: Any) extends RuntimeException {
+final class MatchError(@transient obj: Any) extends RuntimeException {
   /** There's no reason we need to call toString eagerly,
    *  so defer it until getMessage is called.
    */
-  private lazy val objString = {
+  private val objString = {
```

Comment:
The comment would need to be updated here to match the new behaviour.

However, I'm a little concerned about making this change, calling `toString` earlier, it was intentionally deferred in 46d5e73c11bccd0e892429e4c3f2ac1a02cea2a9.

It might be possible to defer `toString` until the point of serialization, with something like this (not tested):

```scala
final class MatchError(@transient obj: Any) extends RuntimeException {
  /** There's no reason we need to call toString eagerly,
   *  so defer it until getMessage is called. We do trigger the message creation on serialization
   *
   */
  private var objString: String = null

  @throws[ObjectStreamException]
  private def writeReplace(): Object = {
    objString = computeObjString
    this
  }

  private def computeObjString: String = {
    def ofClass = "of class " + obj.getClass.getName
    if (obj == null) "null"
    else try {
      obj.toString() + " (" + ofClass + ")"
    } catch {
      case _: Throwable => "an instance " + ofClass
    }
  }

  override def getMessage() = {
    if (objString == null) computeObjString
    objString
  }
}

```